File: leetcode/C++/sequential-digits.cpp
class Solution {
public:
    int makeInitNum(int len){
        string num;
        for(int i=0; i<len; ++i) num.push_back(i+1+'0');
        return stoi(num);
    }
    int makeSumNum(int len){
        string num;
        for(int i=0; i<len; ++i) num.push_back('1');
        return stoi(num);
    }
    vector<int> sequentialDigits(int low, int high) {
        vector<int> sol;
        int mn = to_string(low).length();
        int mx = to_string(high).length();
        for(int i=mn,n=0,p=0; i<=mx && i<10; ++i){
            n=makeInitNum(i); p=makeSumNum(i);
            for(int j=i; j<10; ++j,n+=p) {
                if(n>=low && n<=high) sol.push_back(n);
            }
        }
        return sol;
    }
};

File: leetcode/Python/sequential-digits.py
class Solution:
    def sequentialDigits(self, low: int, high: int) -> List[int]:
        def makeInitNum(len):
            num = ''
            for i in range(len):
                num += str(i+1)
            return int(num)
        def makeSumNum(len):
            num = ''
            for i in range(len):
                num += '1'
            return int(num)
        sol = list()
        mn, mx = len(str(low)), len(str(high))
        for i in range(mn, min(mx+1, 10)):
            n = makeInitNum(i)
            p = makeSumNum(i)
            for j in range(i, 10):
                if n>=low and n<=high:
                    sol.append(n)
                n += p
        return sol